src/main/kotlin/org/rust/cargo/project/CargoProjectDescription.kt

```diff
@@ -142,6 +142,13 @@ class CargoProjectDescription private constructor(
 
     fun findPackage(name: String): Package? = packages.find { it.name == name }
 
+    fun findLibTargetByNormName(normName: String): Pair<Package, Target>? =
+        packages.asSequence()
+            .mapNotNull { pkg ->
+                pkg.libTarget?.let { if (it.normName == normName) Pair(pkg, it) else null }
+            }
+            .firstOrNull()
```

Comment:
I think there are two simplifications possible here. We don't need to return a `Pair<Package, Traget>`, because `Target` has a backling back to package: https://github.com/intellij-rust/intellij-rust/blob/bf18a3c19a8db21a8c925a6996e9255c45ccfe02/src/main/kotlin/org/rust/cargo/project/CargoProjectDescription.kt#L65

Also, this function uses essentially the same algorithm as `findExternCrateRootByName` (list packages, find library targets, filter by normName), so perhaps they can be unified somehow? 

There's `ExternCrate` class nearby, which is similar to `Target`, I think we should use it (and add fields like package name and target). Alternatively, we can jettison `ExternCrate` and just use `Target` instead, with the contract that the kind of target must be `lib`.